src/components/Form.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react'
+import Button from './Button.js'
+import Preview from './forms/Preview'
+import Section from './forms/Section'
+import "../styles/form.css"
+
+export default class Form extends Component {
+
+  constructor( props ) {
+    super( props )
+    this.state = {}
+    this.setFieldState = this.setFieldState.bind( this )
+  }
+
+  componentWillMount() {
+    this.setupState()
+  }
+
+  setupState() {
```

Comment:
The code in this function is too compact - I read it twice and gave up.  Please refactor to make the behavior and purpose explicit.